### DISCUSSION TITLE
Warning:  Encountered two children with the same key

### DISCUSSION THREAD

**Discussion by ostermann-felix** [neuefische/questions#200] [Web] [answered]
Hi friends, I have done alot but while I wanted to finish my last userstory atm i got this error: 

![Bildschirmfoto 2022-09-30 um 16 32 09](https://user-images.githubusercontent.com/94837451/193292967-3891b5d2-b31d-403e-b6b1-208e2610b164.png)

Four same errors with key 0 1 2 3

How can I solve this in my new build where i get every info out of my database?

This is my PR https://github.com/ostermann-felix/capstone-project/pull/9

**Comment by NikRoe** [comment, marked as answer]
Hey Felix,

when mapping over an array you need to ensure that the rendered children have unique keys.
In your `pages/lessons/[id].js` you currently have:
```js
          <>
            <h3 key={index}>{process.name}</h3>
            <p key={index}>{process.instruction}</p>
          </>
```

Your `<h3>` and `<p>` receive the same keys for the moment. 

Those elements either need a parent element that receives a unique key.
Or the keys assigned to those two elements need to differ. For instance:

```js
          <>
            <h3 key={`${index} ${process.name}`}>{process.name}</h3>
            <p key={index}>{process.instruction}</p>
          </>
```
